Src/LiquidProjections/EventMapBuilder.cs

```diff
@@ -150,1125 +118,272 @@ internal void Add(Func<TEvent, TContext, Task> action)
+                    throw new ProjectionException($"Projection {typeof(TProjection)} already exists.");
 
-                private readonly ProjectionEventMappingBuilder<TEvent> eventMappingBuilder;
+                private readonly IAction<TEvent, TContext> actionBuilder;
+                private readonly Func<ProjectorMap<TProjection, TKey, TContext>> projector;
                 private readonly Func<TEvent, TKey> getKey;
 
-                public CreateEventActionBuilder(
-                    ProjectionEventMappingBuilder<TEvent> eventMappingBuilder,
-                    Func<TEvent, TKey> getKey)
+                public CreateAction(IAction<TEvent, TContext> actionBuilder,
+                    Func<ProjectorMap<TProjection, TKey, TContext>> projector, Func<TEvent, TKey> getKey)
                 {
-                    this.eventMappingBuilder = eventMappingBuilder;
+                    this.actionBuilder = actionBuilder;
+                    this.projector = projector;
                     this.getKey = getKey;
                 }
 
-                public void Using(Func<TProjection, TEvent, TContext, Task> projector)
+                public ICreateAction<TEvent, TProjection, TContext> Using(Func<TProjection, TEvent, TContext, Task> projector)
                 {
                     if (projector == null)
                     {
                         throw new ArgumentNullException(nameof(projector));
                     }
 
-                    eventMappingBuilder.innerBuilder.Add((anEvent, context) =>
-                        eventMappingBuilder.eventMapBuilder.projectionModificationHandler(
+                    actionBuilder.As((anEvent, context) => this.projector().Create(
                             getKey(anEvent),
                             context,
                             projection => projector(projection, anEvent, context),
-                            options));
-                }
-            }
-
-            private sealed class CreateIfDoesNotExistEventActionBuilder :
-                ICreateIfDoesNotExistEventActionBuilder<TEvent, TProjection, TContext>
-            {
-                private static readonly ProjectionModificationOptions options = new ProjectionModificationOptions(
-                    MissingProjectionModificationBehavior.Create,
-                    ExistingProjectionModificationBehavior.Ignore);
+                            existingProjection => shouldOverwrite(existingProjection, anEvent, context)));
 
-                private readonly ProjectionEventMappingBuilder<TEvent> eventMappingBuilder;
-                private readonly Func<TEvent, TKey> getKey;
+                    return this;
+                }
 
-                public CreateIfDoesNotExistEventActionBuilder(
-                    ProjectionEventMappingBuilder<TEvent> eventMappingBuilder,
-                    Func<TEvent, TKey> getKey)
+                public ICreateAction<TEvent, TProjection, TContext> IgnoringDuplicates()

```

Comment:
I don't like that you can combine several calls like `.IgnoringDuplicates().OverwritingDuplicates()` in one registration, in which case the last one wins. This also affects the old style API like `.AsCreateIfDoesNotExistOf().OverwritingDuplicates()`.

Reply:
I see that, but it is pretty self-explanatory if you do. Don't feel like fixing that.